drivers/frequency/adf4382/adf4382.c

```diff
@@ -0,0 +1,1303 @@
+	if (ret)
+		return ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x31, ADF4382_EN_ADC_CLK_MSK, 0xff);
+	if (ret)
+		return ret;
+
+	//Time for each VCO calibration decision
+	ret = adf4382_spi_write(dev, 0x37, ADF4382_VCO_CAL_CNT);
+	if (ret)
+		return ret;
+
+	//Time for VCO calibration based on Vtune
+	ret = adf4382_spi_write(dev, 0x38, ADF4382_VCO_CAL_VTUNE);
+	if (ret)
+		return ret;
+
+	//VCO automatic level calibration time
+	ret = adf4382_spi_write(dev, 0x3A, ADF4382_VCO_CAL_ALC);
+	if (ret)
+		return ret;
+
+	tmp = NO_OS_DIV_ROUND_UP(no_os_div_u64(pfd_freq, div1 * 400000) - 2, 4);
+	tmp = no_os_clamp(tmp, 0U, 255U);
+	ret = adf4382_spi_write(dev, 0x3E, tmp);
+	if (ret)
+		return ret;
+
+	// Set LD COUNT
+	ret = adf4382_spi_update_bits(dev, 0x2C, ADF4382_LD_COUNT_OPWR_MSK,
+				      dev->ld_count);
+	if (ret)
+		return ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x2C, ADF4382_LDWIN_PW_MSK,
+				      no_os_field_prep(ADF4382_LDWIN_PW_MSK,
+						      ldwin_pw));
+	if (ret)
+		return ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x11, ADF4382_CLKOUT_DIV_MSK,
+				      no_os_field_prep(ADF4382_CLKOUT_DIV_MSK,
+						      clkout_div));
+	if (ret)
+		return ret;
+
+	val = (n_int >> 8) & ADF4382_N_INT_MSB_MSK;
+	ret = adf4382_spi_update_bits(dev, 0x11, ADF4382_N_INT_MSB_MSK, val);
+	if (ret)
+		return ret;
+
+	// Need to set N_INT last to trigger an auto-calibration
+	val = n_int & ADF4382_N_INT_LSB_MSK;
+	return adf4382_spi_write(dev, 0x10, val);
+}
+
+/**
+ * @brief Set the phase adjustment in pico-seconds. The phase adjust will
+ * enable the Bleed current option as well as delay mode to 0.
+ * @param dev 		- The device structure.
+ * @param phase_ps 	- The phase adjustment in pico-seconds.
+ * @return    		- 0 in case of success, negative error code otherwise.
+ */
+int adf4382_set_phase_adjust(struct adf4382_dev *dev, uint32_t phase_ps)
+{
+	uint64_t phase_reg_value;
+	uint32_t rfout_deg_ns;
+	uint32_t phase_deg_ns;
+	uint64_t rfout_deg_s;
+	uint64_t phase_bleed;
+	uint16_t phase_deg;
+	uint64_t phase_ci;
+	uint64_t pfd_freq;
+	int ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x1E, ADF4382_EN_PHASE_RESYNC_MSK, 0xff);
+	if(ret)
+		return ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x1F, ADF4382_EN_BLEED_MSK, 0xff);
+	if(ret)
+		return ret;
+
+	ret = adf4382_spi_update_bits(dev, 0x32, ADF4382_DEL_MODE_MSK, 0x0);
+	if(ret)
+		return ret;
+
+	dev->phase_adj = phase_ps;
+
+	//Determine the output freq. in degrees/s
+	rfout_deg_s = 360 * dev->freq;
+	//Convert it to degrees/ns
+	rfout_deg_ns = no_os_div_u64(rfout_deg_s, S_TO_NS);
+	//Determine the phase adjustment in degrees relative the output freq.
+	phase_deg_ns = rfout_deg_ns * phase_ps;
+	//Convert it to degrees/ps
+	phase_deg = no_os_div_u64(phase_deg_ns, NS_TO_PS);
+
+	if(phase_deg > 360) {
+		pr_err("Phase Adjustment cannot exceed 360deg per Clock Period\n");
+		return EINVAL;
+	}
+
+	/*Phase adjustment can only be done if bleed is active, and a bleed
```

Comment:
there is some inconsistency here in how you wrote comments with // or with /*, please make them look alike